evm/src/executor/inspector/fuzzer.rs

```diff
@@ -0,0 +1,118 @@
+use crate::{
+    fuzz::{invariant::RandomCallGenerator, strategies::EvmFuzzState},
+    utils, TEST_CONTRACT_ADDRESS,
+};
+use bytes::Bytes;
+use revm::{db::Database, CallInputs, CallScheme, EVMData, Gas, Inspector, Interpreter, Return};
+
+/// An inspector that can fuzz and collect data for that effect.
+#[derive(Clone, Debug)]
+pub struct Fuzzer {
+    /// Given a strategy, it generates a random call.
+    pub call_generator: Option<RandomCallGenerator>,
+    /// If set, it collects `stack` and `memory` values for fuzzing purposes.
+    pub collect: bool,
+    /// If `collect` is set, we store the collected values in this fuzz dictionary.
+    pub fuzz_state: EvmFuzzState,
```

Comment:
Going through the file it seems that this sort of does double work in the sense that the "normal"/current fuzzer already does this outside of the executor?

Reply:
https://github.com/foundry-rs/foundry/pull/1572#issuecomment-1186479575